src/core/emergency-action.c

```diff
@@ -28,10 +40,23 @@ void emergency_action(
                 int exit_status,
                 const char *reason) {
 
+        Unit *u;
+
         assert(m);
         assert(action >= 0);
         assert(action < _EMERGENCY_ACTION_MAX);
 
+        /* Is the special shutdown target active or queued? If so, we are in shutdown state */
+        u = manager_get_unit(m, SPECIAL_SHUTDOWN_TARGET);
+        if (u && unit_active_or_pending(u) && IN_SET(action,
+                                                     EMERGENCY_ACTION_REBOOT,
+                                                     EMERGENCY_ACTION_POWEROFF,
+                                                     EMERGENCY_ACTION_EXIT)) {
+                log_notice("Shutdown is already active. Skipping emergency action request %s.",
+                           emergency_action_table[action]);
+                return;
+        }
```

Comment:
hmm, can you reverse the check? manager_get_unit() is more expensive than just comparing the action. hence maybe do it like this:

```c
if (IN_SET(action, EMERGENCY_ACTION_REBOOT, EMERGENCY_ACTTION_POWEROFF, EMERGENCY_ACTION_EXIT)) {
        Unit *u;
        u = manager_get_unit(m, …);
        if (u && unit_active_or_pending(u)) {
                …
```

Reply:
This is done in latest PR